spec/features/idv/doc_auth/document_capture_step_spec.rb

```diff
@@ -314,6 +320,19 @@
         expect(page).to have_current_path(next_step)
       end
     end
+
+    context 'when using async uploads', :js do
+      let(:document_capture_async_uploads_enabled) { true }
+      let(:doc_auth_enable_presigned_s3_urls) { true }
+
+      it 'proceeds to the next page with valid info' do
+        mock_document_capture_result
+        attach_images(liveness_enabled: false)
+        click_on 'Submit'
+
+        expect(page).to have_current_path(next_step, wait: 20)
```

Comment:
WDYT of adding some tests that check inside the `Test::FakeS3Controller.data` hash to make sure stuff is there?

something like....


```ruby
before do
  Test::FakeS3Controller.clear!
end

it '...' do
  expect(Test::FakeS3Controller.data).to be_empty

  mock_document_capture_result
  attach_images(liveness_enabled: false)
  click_on 'Submit'

  expect(Test::FakeS3Controller.data.size).to eq(2) # or 3 if we do selfie?
end
```

Reply:
> WDYT of adding some tests that check inside the `Test::FakeS3Controller.data` hash to make sure stuff is there?

Yeah, I'd wondered about that, and initially thought since we [assert this in the controller test](https://github.com/18F/identity-idp/blob/547d1bdb39ae2c129f272a045c4e95b244f64161/spec/controllers/fake_s3_controller_spec.rb#L38-L41) it may not be necessary to do here, though now that you mention it, I think it's a good extra assurance it works as we expect in a more "real" exercise.

Reply:
Another option if you don't want to dig "inside" would be to `GET` from the same endpoint and read the data back

Reply:
> Another option if you don't want to dig "inside" would be to `GET` from the same endpoint and read the data back

Seems like a nice idea to complete the circle! I had to do a bit of a Capybara driver dance to verify that the resource exists, but otherwise pretty straightforward.

See 5e8e6cc